@here/harp-mapview/lib/geometry/TileGeometryManager.ts

```diff
@@ -193,23 +193,15 @@ export abstract class TileGeometryManagerBase implements TileGeometryManager {
         addOrRemoveToHiddenSet: boolean = true
     ): void {
         let visibilityHasChanged = false;
-        if (Array.isArray(kind)) {
-            for (const oneKind of kind as GeometryKind[]) {
-                visibilityHasChanged =
-                    visibilityHasChanged ||
-                    this.addRemove(this.hiddenKinds, oneKind, addOrRemoveToHiddenSet);
-            }
-        } else if (kind instanceof Set) {
-            const kindSet = kind as GeometryKindSet;
-            for (const oneKind of kindSet) {
+
+        if (Array.isArray(kind) || kind instanceof Set) {
+            for (const oneKind of kind) {
                 visibilityHasChanged =
-                    visibilityHasChanged ||
-                    this.addRemove(this.hiddenKinds, oneKind, addOrRemoveToHiddenSet);
+                    this.addRemove(this.hiddenKinds, oneKind, addOrRemoveToHiddenSet) ||
+                    visibilityHasChanged;
```

Comment:
I would even go further, create temp bool as result of `addRemove` and then make boolean alternative.
```
const visibilityChange = this.addRemove(this.hiddenKinds, oneKind, addOrRemoveToHiddenSet)
visibilityHasChanged = visibilityHasChanged || visibilityChange;
```
This is assuming that we want always call "addRemove". Previous code (and even) current, may be prone for optimization, I am not sure if any release of JIT compiler may ready it in revese order, as least I don't know if it is writtern in any standard, so we may do in cleaner and for sure.

Reply:
done.